lambdas/send_from_queue/send_from_queue.py

```diff
@@ -208,19 +208,34 @@ def lambda_handler(event, _context):
                     "queue. {}".format(message_body)
                 )
 
+            api_url_full_query = api_url
+            already_triggered = message_body.get("already_triggered")
+            if already_triggered:
+                    api_url_full_query = api_url_full_query + \
+                        "&already_triggered={}".format(already_triggered)
+
             # Send the SQS message to the API Gateway
             response = post(
-                api_url_query,
+                api_url_full_query,
                 json=payload,
                 headers=headers,
                 timeout=(3.05, 30)
             )
 
-            # If there was a problem, raise an error
-            response.raise_for_status()
-
-            # Otherwise, delete the message since it has been processed
-            _delete_from_queue(queue_object, message)
+            error_message = "More jobs triggered, but " \
+                            "unable to trigger all jobs."
+            if error_message in response.text:
+                # The hook failed to trigger all jobs, but it has
+                # been replaced in the queue with a new hook, so
+                # delete this one.
+                _delete_from_queue(queue_object, message)
+                response.raise_for_status()
+            else:
+                # If there was a problem, raise an error
+                response.raise_for_status()
+
+                # Otherwise, delete the message since it has been processed
+                _delete_from_queue(queue_object, message)
```

Comment:
If the `error_message` defined is in the response then that means the hook was sent from the queue and triggered more jobs than it had previously (lets say it only triggered `edx-platform-js-pr` the first time, but this time `edx-platform-quality-pr` was triggered), then we delete this queue message before raising the error. This is because an updated hook has been stored in the queue. 
If there is an error and it was not `error_message` then the else statement ensures the messages does not get deleted.
Does that make sense? The difference is whether or not the queue message gets deleted when there is an error.